src/Build/Evaluation/Evaluator.cs

```diff
@@ -1256,7 +1256,7 @@ private ICollection<P> AddEnvironmentProperties()
 
             foreach (ProjectPropertyInstance environmentProperty in _environmentProperties)
             {
-                P property = _data.SetProperty(environmentProperty.Name, ((IProperty)environmentProperty).EvaluatedValueEscaped, false /* NOT global property */, false /* may NOT be a reserved name */);
+                P property = _data.SetProperty(environmentProperty.Name, ((IProperty)environmentProperty).EvaluatedValueEscaped, false /* NOT global property */, false /* may NOT be a reserved name */, true /* IS environment variable. */);
```

Comment:
>/* IS environment variable. * [](start = 207, length = 29)

nit: you could use named args

Reply:
It was less change to just add onto the method call.